low-level/files/opt/seagate/sspl/bin/sspl_setup

```diff
@@ -1,97 +1,234 @@
-        logger -i -p local3.err "SSPL is not configured. Run provisioner scripts in $SSPL_BASE_DIR/bin."
-        exit 1
-        ;;
-
-    * )
-        usage
-        ;;
-esac
+import sys
+import errno
+import argparse
+import inspect
+import traceback
+import os
+import subprocess
+# Add the top level directories
+sys.path.insert(0, '/opt/seagate/cortx/sspl/low-level')
+from framework.base.sspl_constants import PRODUCT_FAMILY
+from files.opt.seagate.sspl.bin import validate_consul_config
+
+class Cmd:
+    """Setup Command"""
+
+    def __init__(self, args: dict):
+        self._args = args.args
+        self._script_dir = os.path.dirname(os.path.abspath(__file__))
+
+    @property
+    def args(self) -> str:
+        return self._args
+
+    @staticmethod
+    def usage(prog: str):
+        """Print usage instructions"""
+
+        sys.stderr.write(
+            f"{prog} [post_install[-p <LDR_R1>]|init [-dp] [-r <ssu|gw|cmu|vm>]\n|config [-f] [-r <ssu|gw|cmu|vm>]|test [self|sanity]|reset [hard -p <LDR_R1>|soft]]\n"
+            "init options:\n"
+            "\t -dp Create configured datapath\n"
+            "\t -r  Role to be configured on the current node\n"
+            "config options:\n"
+            "\t -f  Force reinitialization. Do not prompt\n"
+            "\t -r  Role to be configured on the current node"
+            "post_install options:\n"
+            "\t -p Product to be configured\n"
+            "reset options:\n"
+            "\t -p Product to be configured\n")
+
+    @staticmethod
+    def get_command(desc: str, argv: dict):
+        """Return the Command after parsing the command line."""
+
+        parser = argparse.ArgumentParser(desc)
+        subparsers = parser.add_subparsers()
+        cmds = inspect.getmembers(sys.modules[__name__])
+        cmds = [(x, y) for x, y in cmds
+            if x.endswith("Cmd") and x != "Cmd"]
+        for name, cmd in cmds:
+            cmd.add_args(subparsers, cmd, name)
+
+        args, unknown = parser.parse_known_args(argv)
+        args.args = unknown + args.args
+        return args.command(args)
+
+    @staticmethod
+    def add_args(parser: str, cls: str, name: str):
+        """Add Command args for parsing."""
+
+        parsers = parser.add_parser(cls.name, help='setup %s' % name)
+        parsers.add_argument('args', nargs='*', default=[], help='args')
+        parsers.set_defaults(command=cls)
+
+    @staticmethod
+    def _send_command(command: str, fail_on_error=True):
+        process = subprocess.Popen(
+            command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        response, error = process.communicate()
+        if error is not None and \
+        len(error) > 0:
+            print("command '%s' failed with error\n%s" % (command, error))
+            if fail_on_error:
+                sys.exit(1)
+            else:
+                return str(error)
+        return str(response)
+
+    # @staticmethod
```

Comment:
keep for now. but remove if not used, from final commit or second commit for this task(importing required classes for other sspl setup classes like post install, config, init, etc)

Reply:
Okay